File: client/src/app/_models/userParams.ts
import { ILoggedUser } from "./loggedUser";


export class UserParams {
  gender: string;
  minAge = 18;
  maxAge = 90;
  pageNumber = 1;
  pageSize = 6;
  orderBy = 'lastActive';

  constructor(user: ILoggedUser) {
    this.gender = user.gender === 'female' ? 'male' : 'female';
  }
}

File: API/Helpers/Params/UserParams.cs
using API.Helpers.Pagination;

namespace API.Helpers.Params
{
    public class UserParams : PageParams
    {
        public string CurrentUsername { get; set; }
        public string Gender { get; set; }
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 90;
        public string OrderBy { get; set; } = "lastActive";
    }
}
